cmd/plugins.go

```diff
@@ -202,34 +214,47 @@ var pluginTidyCmd = &cobra.Command{
 				}
 
 				if !pluginInUse || tidyPurge {
-					if dryRun {
-						log.Infof("       plugin: [%s] can be removed", pluginName)
-						continue
-					}
-					pluginImportPath := filepath.Join(desktopCmdPath, f.Name())
-
 					// clean-up go.mod
-					pluginImportStr, _ := readPluginGoImport(pluginImportPath, pluginName)
+					pluginImportStr, err := readPluginGoImport(pluginImportPath, pluginName)
+
 					// Delete the 'replace' and 'require' import strings from go.mod.
 					// Not mission critical, if the plugins not correctly removed from
 					// the go.mod file, the project still works and the plugin is
 					// successfully removed from the flutter.Application.
 					if err != nil || pluginImportStr == "" {
 						log.Warnf("Couldn't clean the '%s' plugin from the 'go.mod' file. Error: %v", pluginName, err)
-					} else {
-						fileutils.RemoveLinesFromFile(filepath.Join(build.BuildPath, "go.mod"), pluginImportStr)
+					} else if err = gomod.DropRequire(pluginImportStr); err != nil {
```

Comment:
DropRequire is not idempotent

Reply:
not following. removing an item from a set is always idempotent.

are you referring to some other state?

Reply:
I mean that the set is modified, so the inlined statement is not just there to obtain information necessary for the if block. Generally I never inline those statements in the `if` line because it reads backwards, especially with else-if.

Reply:
are you asking that I just move it inside the else block? I don't mind moving it.

or are you referring to the fact that previously fileutils kills the process if it can't modify the file? I can readd that behavior instead of warning.

but neither of those situations are in any way related to idempotent behavior.

Reply:
I was misusing the term, sorry. I meant to move it inside the else block, yes.